ledger/db/src/lib.rs

```diff
@@ -470,6 +524,13 @@ impl LedgerDB {
             }
         }
 
+        // Check that none of the output public keys appear in the ledger.
+        for output in block_contents.outputs.iter() {
+            if self.contains_tx_out_public_key(&output.public_key)? {
```

Comment:
There is a potential problem with this and the aggregate fee output. It is possible for an attacker to force a collision by creating an output in the ledger with a particular public key, and then forcing the enclave to use that public key as the aggregate fee's output public key. The attack is pretty contrived, but here it is: